Vungle/src/main/java/com/mopub/nativeads/VungleAdRenderer.java

```diff
@@ -0,0 +1,357 @@
+        if (vungleNativeViewHolder.mainView instanceof NativeAdLayout) {
+            nativeAdLayout = (NativeAdLayout) vungleNativeViewHolder.mainView;
+        } else {
+            nativeAdLayout = new NativeAdLayout(context);
+            insertVungleNativeAdLayout(nativeAdLayout, vungleNativeViewHolder.mainView);
+        }
+
+        RelativeLayout adChoicesContainer = vungleNativeViewHolder.getAdChoicesContainer();
+        if (adChoicesContainer != null) {
+            adChoicesContainer.removeAllViews();
+            final NativeAdOptionsView adOptionsView =
+                    new NativeAdOptionsView(
+                            adChoicesContainer.getContext(),
+                            nativeAd.getVungleNativeAd(),
+                            nativeAdLayout);
+            ViewGroup.LayoutParams layoutParams = adOptionsView.getLayoutParams();
+            if (layoutParams instanceof RelativeLayout.LayoutParams) {
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
+                    ((RelativeLayout.LayoutParams) layoutParams).addRule(
+                            RelativeLayout.ALIGN_PARENT_END);
+                } else {
+                    ((RelativeLayout.LayoutParams) layoutParams).addRule(
+                            RelativeLayout.ALIGN_PARENT_RIGHT);
+                }
+            }
+            adChoicesContainer.addView(adOptionsView);
+        }
+
+        nativeAd.registerChildViewsForInteraction(nativeAdLayout,
+                vungleNativeViewHolder.getMediaView(), vungleNativeViewHolder.getAdIconView());
+    }
+
+    private void insertVungleNativeAdLayout(NativeAdLayout nativeAdLayout, View mainView) {
+        MoPubLog.log(SHOW_ATTEMPTED, ADAPTER_NAME);
+
+        if (mainView instanceof FrameLayout) {
+            FrameLayout outerFrame = (FrameLayout) mainView;
+            View actualView = outerFrame.getChildAt(0);
+
+            nativeAdLayout.setLayoutParams(new ViewGroup.LayoutParams(
+                    ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT));
+
+            outerFrame.removeView(actualView);
+            nativeAdLayout.addView(actualView);
+            outerFrame.addView(nativeAdLayout);
+        } else {
+            MoPubLog.log(CUSTOM, ADAPTER_NAME, "Couldn't add Vungle native ad view. Wrapping view not found.");
+        }
+    }
+
+    @Override
+    public boolean supports(@NonNull BaseNativeAd nativeAd) {
+        Preconditions.checkNotNull(nativeAd);
+        return nativeAd instanceof VungleNative.VungleNativeAd;
+    }
+
+    static class VungleNativeViewHolder {
+        @Nullable
+        private View mainView;
+        @Nullable
+        private TextView titleView;
+        @Nullable
+        private TextView textView;
+        @Nullable
+        private TextView callToActionView;
+        @Nullable
+        private ImageView adIconView;
+        @Nullable
+        private TextView mStarRatingTextView;
```

Comment:
subjective, but do we need "m" prefix? especially considering all  other member fields don't have it